src/lib/storage/storage_manager.hpp

```diff
@@ -94,6 +94,14 @@ class StorageManager : public Noncopyable {
   std::shared_ptr<Chunk> map_chunk_from_disk(const uint32_t chunk_offset_end, const std::string& filename,
                                              const uint32_t segment_count);
 
+  CHUNK_HEADER read_chunk_header(const std::string& filename, const uint32_t segment_count,
+                                 const uint32_t chunk_offset_begin);
+
+  std::vector<uint32_t> generate_segment_offset_ends(const std::shared_ptr<Chunk> chunk);
+  void write_dict_segment_to_disk(const std::shared_ptr<DictionarySegment<int>> segment, const std::string& file_name);
+  void write_chunk_to_disk(const std::shared_ptr<Chunk>& chunk, const std::vector<uint32_t>& segment_offset_ends,
+                           const std::string& file_name);
+
```

Comment:
these functions can stay private by adding StorageManager as friend to each test class. Tobi knows how to do this or you check the E2E test of [file_format_refactored](https://github.com/hyrise-mp-22-23/hyrise/tree/file_format_refactored) where I've done this with a constant :)